problems/basic.go

```diff
@@ -0,0 +1,78 @@
+package problems
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"go.opencensus.io/trace"
+)
+
+// BasicProblem, common fields for most Problems. Useful for embedding into
+// for custom problem types.
+type BasicProblem struct {
+	// Type, a URI reference that identifies the problem type.
+	// When dereferenced this should provide human-readable documentation for the
+	// problem type. When member is not present it is assumed to be "about:blank".
+	Type string `json:"type" format:"url"`
+
+	// Title, a short, human-readable summary of the problem type.
+	// This should always be the same value for the same Type.
+	Title string `json:"title"`
+
+	// Status, the HTTP status code associated with this problem occurrence.
+	Status int `json:"status,omitempty"`
+
+	// Detail, a human-readable explanation specific to this occurrence of the problem.
+	Detail string `json:"detail,omitempty"`
+
+	// Instance, a URI reference that identifies the specific occurrence of the problem.
```

Comment:
Should this be "identifies the specific resource on which the problem occurred"?